src/platform/datapath_raw.h

```diff
@@ -273,7 +256,9 @@ typedef struct CXPLAT_SOCKET {
 
     CXPLAT_SEND_DATA* PausedTcpSend; // Paused TCP send data *before* framing
     CXPLAT_SEND_DATA* CachedRstSend; // Cached TCP RST send data *after* framing
-} CXPLAT_SOCKET;
+
+    CXPLAT_SOCKET;
```

Comment:
Generally, it's best to put these up front, because then it makes casting much easier.

Reply:
Sadly impossible as CXPLAT_SOCKET has variable length member `CXPLAT_SOCKET_PROC PerProcSockets[0]`.
Move the member to `CXPLAT_SOCKET_RAW` from `CXPLAT_SOCKET`?